cmd/src/debug_kube.go

```diff
@@ -0,0 +1,389 @@
+			ch <- getSiteConfig(ctx, baseDir)
+		}()
+	}
+
+	// close channel when wait group goroutines have completed
+	go func() {
+		wg.Wait()
+		close(ch)
+	}()
+
+	// write to archive all the outputs from kubectl call functions passed to buffer channel
+	for f := range ch {
+		if f.err != nil {
+			log.Printf("getting data for %s failed: %v\noutput: %s", f.name, f.err, f.data)
+			continue
+		}
+
+		if verbose {
+			log.Printf("archiving file %q with %d bytes", f.name, len(f.data))
+		}
+
+		zf, err := zw.Create(f.name)
+		if err != nil {
+			return fmt.Errorf("failed to create %s: %w", f.name, err)
+		}
+
+		_, err = zf.Write(f.data)
+		if err != nil {
+			return fmt.Errorf("failed to write to %s: %w", f.name, err)
+		}
+	}
+
+	return nil
+}
+
+func selectPods(ctx context.Context, namespace string) (podList, error) {
+	// Declare buffer type var for kubectl pipe
+	var podsBuff bytes.Buffer
+
+	// Get all pod names as json
+	podsCmd := exec.CommandContext(
+		ctx,
+		"kubectl", "-n", namespace, "get", "pods", "-l", "deploy=sourcegraph", "-o=json",
+	)
+	podsCmd.Stdout = &podsBuff
+	podsCmd.Stderr = os.Stderr
+	err := podsCmd.Run()
+	if err != nil {
+		fmt.Errorf("failed to aquire pods for subcommands with err: %v", err)
+	}
+
+	//Decode json from podList
+	var pods podList
+	if err := json.NewDecoder(&podsBuff).Decode(&pods); err != nil {
+		fmt.Errorf("failed to unmarshall get pods json: %w", err)
+	}
+
+	return pods, err
+}
+
+func getPods(ctx context.Context, namespace, baseDir string) *archiveFile {
+	return archiveFileFromCommand(
+		ctx,
+		filepath.Join(baseDir, "/kubectl/getPods.txt"),
+		"kubectl", "-n", namespace, "get", "pods", "-o", "wide",
+	)
+}
+
+func getEvents(ctx context.Context, namespace, baseDir string) *archiveFile {
+	return archiveFileFromCommand(
+		ctx,
+		filepath.Join(baseDir, "/kubectl/events.txt"),
+		"kubectl", "-n", namespace, "get", "events",
+	)
+}
+
+func getPV(ctx context.Context, namespace, baseDir string) *archiveFile {
+	return archiveFileFromCommand(
+		ctx,
+		filepath.Join(baseDir, "/kubectl/persistent-volumes.txt"),
+		"kubectl", "-n", namespace, "get", "pv",
+	)
+}
+
+func getPVC(ctx context.Context, namespace, baseDir string) *archiveFile {
+	return archiveFileFromCommand(
+		ctx,
+		filepath.Join(baseDir, "/kubectl/persistent-volume-claims.txt"),
+		"kubectl", "-n", namespace, "get", "pvc",
+	)
+}
+
+// get kubectl logs for pod containers
+func getPodLog(ctx context.Context, podName, containerName, namespace, baseDir string) *archiveFile {
+	return archiveFileFromCommand(
+		ctx,
+		filepath.Join(baseDir, "/kubectl/pods/", podName, containerName)+".log",
+		"kubectl", "-n", namespace, "logs", podName, "-c", containerName,
+	)
+}
+
+// get kubectl logs for past container
+func getPastPodLog(ctx context.Context, podName, containerName, namespace, baseDir string) *archiveFile {
+	return archiveFileFromCommand(
+		ctx,
+		filepath.Join(baseDir, "/kubectl/pods/", podName, "prev-"+containerName)+".log",
```

Comment:
(Some other occurrences of the filename construction I've noted above)

(Also general advice, sometimes reviewers will only flag one occurrence of an issue but will implicitly mean "fix this wherever it occurs" - that doesn't always happen though because it's easy to forget to scan through all the relevant code and find everything)

Reply:
I messed this up first go around and was using `path` instead of `path/filepath` -- for some reason I ended up being unable to join some of the paths correctly and maintain the file extension at the end. This resulted in a bunch of empty dirs being archived. Will give this another go around based on follow-up feedback.